plugins/modules/dw_dbc.py

```diff
@@ -0,0 +1,178 @@
+        state.
+    type: int
+    required: False
+    default: 3600
+    aliases:
+      - polling_timeout  
+extends_documentation_fragment:
+  - cloudera.cloud.cdp_sdk_options
+  - cloudera.cloud.cdp_auth_options
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details.
+
+# Create Database Catalog
+- cloudera.cloud.dw_dbc:
+    name: example-database-catalog
+    cluster_id: example-cluster-id
+'''
+
+RETURN = r'''
+---
+dbcs:
+  description: The information about the named Database Catalog.
+  type: list
+  returned: always
+  elements: complex
+  contains:
+    id:
+      description: The id of the Database Catalog.
+      returned: always
+      type: str
+    name:
+      description: The name of the Database Catalog.
+      returned: always
+      type: str
+    status:
+      description: The status of the Database Catalog.
+      returned: always
+      type: str
+sdk_out:
+  description: Returns the captured CDP SDK log.
+  returned: when supported
+  type: str
+sdk_out_lines:
+  description: Returns a list of each line of the captured CDP SDK log.
+  returned: when supported
+  type: list
+  elements: str
+'''
+
+
+class DwDbc(CdpModule):
+    def __init__(self, module):
+        super(DwDbc, self).__init__(module)
+
+        # Set variables
+        self.cluster_id = self._get_param('cluster_id')
+        self.name = self._get_param('name')
+        self.load_demo_data = self._get_param('load_demo_data')
+        self.state = self._get_param('state')
+        self.wait = self._get_param('wait')
+        self.delay = self._get_param('delay')
+        self.timeout = self._get_param('timeout')
+
+        # Initialize return values
+        self.dbcs = []
+
+        # Execute logic process
+        self.process()
+
+    @CdpModule._Decorators.process_debug
+    def process(self):
+        self.name = self.cdpy.dw.create_dbc(cluster_id=self.cluster_id, name=self.name,
+                                            load_demo_data=self.load_demo_data)
+        if self.wait:
+            self.target = self.cdpy.sdk.wait_for_state(
+                describe_func=self.cdpy.dw.describe_dbc,
+                params=dict(cluster_id=self.cluster_id, dbc_id=self.name['dbcId']),
+                state='Running', delay=self.delay, timeout=self.timeout
+            )
+        else:
+            self.target = self.cdpy.dw.describe_dbc(cluster_id=self.cluster_id, dbc_id=self.name['dbcId'])
+        self.dbcs.append(self.target)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=CdpModule.argument_spec(
+            cluster_id=dict(required=True, type='str', aliases=['cluster_id']),
+            name = dict(required=True, type='str', aliases=['name']),
+            load_demo_data=dict(required=False, type='bool', aliases=['load_demo_data']),
+            state=dict(required=False, type='str', choices=['present', 'absent'], default='present'),
```

Comment:
Please implement the absent state teardown controls, and make the module idempotent